include/common_structs.h

```diff
@@ -4,6 +4,12 @@
 #include "ultra64.h"
 #include "types.h"
 
+typedef struct vec3f {
```

Comment:
hm, this will be a conflict with #18

Reply:
I can resolve the conflicts if #21 is merged before, that's fine.
vec3f and npc are lifted from my commit anyways.

Reply:
I'm happy to merge this one first and then resolve it

Reply:
Discussed on discord; alex is planning to merge this into their PR, so no need to worry here.

Reply:
It shouldn't be hard to merge. I just needed those structs and copied them over.